homeassistant/components/geniushub/switch.py

```diff
@@ -0,0 +1,51 @@
+"""Support for Genius Hub switch/outlet devices."""
+from homeassistant.components.switch import SwitchDevice
+from homeassistant.helpers.typing import ConfigType, HomeAssistantType
+
+from . import DOMAIN, GeniusZone
+
+GH_ZONE_ATTR = "on / off"
+
+
+async def async_setup_platform(
+    hass: HomeAssistantType, config: ConfigType, async_add_entities, discovery_info=None
+) -> None:
+    """Set up the Genius Hub switch entities."""
+    if discovery_info is None:
+        return
+
+    broker = hass.data[DOMAIN]["broker"]
+
+    switches = [
+        GeniusSwitch(broker, z)
+        for z in broker.client.zone_objs
+        if GH_ZONE_ATTR in z.data["type"]
+    ]
+
+    async_add_entities(switches, update_before_add=True)
+
+
+class GeniusSwitch(GeniusZone, SwitchDevice):
+    """Representation of a Genius Hub switch."""
+
+    @property
+    def is_on(self) -> bool:
+        """Return the status of the sensor."""
+        # off, override, timer, footprint
+        # technially this could be untrue, if we're using the timer
+        return self._zone.data["mode"] != "off"
```

Comment:
It is either (these are both boolean):
```python
self._zone.data["_state"].get("bIsActive"):
```
... or:
```
self._zone.data["_state"].get("bOutRequestHeat"):
```
Sorry, I can't recall exactly, but I think it is the second - please check.

Reply:
Ah, _state is the unchanged v3 response? On my outlet zone which is currently on, it responds

`            "bHeatEnabled": true,
            "bIsActive": true,
            "bOutRequestHeat": false,
            "bPrimaryZone": true,
`

Reply:
Yes, must be `bIsActive`, but actually, I just realized that this limits us to the v3 API - this is inappropriate.

... let me think on it.

Reply:
> The problem is, and this is why I haven't been rushing to implement this functionality, that HA switches have no concept of operating mode / profile. That is, they expect to be turned off, and turned on - they have no concept of following a schedule (Timer mode)...

Ahh fun, yes does look like we'd need to expand bits of HA or create a new base thing, to support switches with timers etc. I figured it was nice to have something that at least worked .. perfect being the enemy of the good, and all that.

btw would you take a patch to ghclient repo, for the "set_mode("override", duration) without temp being mentioned ? seems this is probably required for it to stay on..

Reply:
> btw would you take a patch to ghclient repo, for the "set_mode("override", duration) without temp being mentioned ? seems this is probably required for it to stay on..

All contributions welcome!